Detectors/MUON/MID/Simulation/test/testSimulation.cxx

```diff
@@ -0,0 +1,208 @@
+      if (isFired != (currStrip >= firstStrip && currStrip <= lastStrip)) {
+        isOk = false;
+        ss << "Cathode: " << cathode << "  firstStrip: " << firstStrip << "  lastStrip: " << lastStrip;
+        ss << "  line: " << iline << "  strip: " << istrip << "  fired: " << isFired << "\n";
+      }
+    }
+  }
+  errorMessage = ss.str();
+  return isOk;
+} // namespace mid
+
+std::vector<int> getDEList()
+{
+  std::vector<int> deList = { 2, 3, 4, 5, 6, 20, 21, 22, 23, 24, 47, 48, 49, 50, 51, 65, 66, 67, 68, 69 };
+  return deList;
+}
+
+BOOST_TEST_GLOBAL_FIXTURE(SIMUL);
+
+BOOST_AUTO_TEST_SUITE(o2_mid_simulation)
+// BOOST_FIXTURE_TEST_SUITE(sim, SIMUL)
+
+BOOST_DATA_TEST_CASE(MID_SimulChain, boost::unit_test::data::make(getDEList()), deId)
+{
+  // In this test, we generate a cluster from one impact point and we reconstruct it.
+  // If the impact point is in the RPC, the digitizer will return a list of strips,
+  // that are close to each other by construction.
+  // We will therfore have exactly one reconstructed cluster.
+  // It is worth noting, however, that the clustering algorithm is desigend to
+  // produce two clusters if the BP and NBP do not superpose.
+  // The digitizer produces superposed BP and NBP strips only if the response parameters
+  // are the same for both.
+  // Otherwise we can have from 1 to 3 clusters produced.
+
+  std::vector<ColumnDataMC> digitStoreMC;
+  o2::dataformats::MCTruthContainer<MCLabel> digitLabelsMC;
+  std::vector<o2::mid::ColumnData> digitStore;
+  o2::dataformats::MCTruthContainer<MCLabel> digitLabels;
+
+  for (int ievent = 0; ievent < 100; ++ievent) {
+    auto hits = generateHits(1, deId, SIMUL::mapping, SIMUL::geoTrans);
+    std::stringstream ss;
+    int nGenClusters = 1, nRecoClusters = 0;
+    SIMUL::digitizer.process(hits, digitStoreMC, digitLabelsMC);
+    SIMUL::digitsMerger.process(digitStoreMC, digitLabelsMC, digitStore, digitLabels);
+    SIMUL::preClusterizer.process(digitStore);
+    SIMUL::clusterizer.process(SIMUL::preClusterizer.getPreClusters());
+    nRecoClusters = SIMUL::clusterizer.getNClusters();
+    ss << "nRecoClusters: " << nRecoClusters << "  nGenClusters: " << nGenClusters << "\n";
+    for (auto& col : digitStore) {
+      ss << col << "\n";
+    }
+    ss << "\n  Clusters:\n";
+    for (auto it = SIMUL::clusterizer.getClusters().begin(); it < SIMUL::clusterizer.getClusters().begin() + nRecoClusters; ++it) {
+      ss << "pos: (" << it->xCoor << ", " << it->yCoor << ")  sigma2: (" << it->sigmaX2 << ", " << it->sigmaY2 << ")\n";
+    }
+
+    int nColumns = digitStore.size();
+
+    if (SIMUL::params.getParB(0, deId) == SIMUL::params.getParB(1, deId) && nColumns <= 2) {
+      BOOST_TEST((nRecoClusters == nGenClusters), ss.str());
+    } else {
+      BOOST_TEST((nRecoClusters >= nGenClusters && nRecoClusters <= nColumns), ss.str());
+    }
+  }
+}
+
+BOOST_DATA_TEST_CASE(MID_Digitizer, boost::unit_test::data::make(getDEList()), deId)
+{
```

Comment:
missing a comment here ...

also, like the previous test case, this one is doing too much IMO, i.e. it's far from a "unit" test.

Reply:
The test is simply checking that the labels are correctly assigned. Why do you say it is doing too much?
I can split the digitizer and digit-merger parts...but I need the digitizer to build the digits to be merged, so I cannot really split it in two...unless I re-write the digitizer part without testing it when I test the digit merger...

Reply:
Warning, I'm starting to get pedantic here, I know ;-) 
The idea is that tests follow a regular pattern and thus are easy to grasp for the reader : first the elements required for the test setup, then what behaviour is executed, and lastly how that behaviour is verified. (steps known as Arrange,Act,Assert or Setup, Execute, Verify, or Given,When,Then in TDD methodology).
The MID_Digitizer does not really follow this simple pattern as you have several BOOT_TEST scattered.
-end of pedantic part | discard this whole comment if you want ;-) -